src/Protector.php

```diff
@@ -385,7 +386,10 @@ protected function generateDump(array $options = []): ?string
         $dumpOptions->push(sprintf('-p%s', escapeshellarg($this->connectionConfig['password'])));
         $dumpOptions->push(sprintf('--max-allowed-packet=%s', escapeshellarg(config('protector.maxPacketLength'))));
         $dumpOptions->push('--no-create-db');
-        $dumpOptions->push('--set-gtid-purged=off');
+
+        if (!DB::connection()->isMaria()) {
```

Comment:
for some reason, this wonderful little line of code leads to the following failure codes in docker

SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for laravel-protector-mysql_testing-1 failed: Name or service not known

SQLSTATE[HY000] [2002] php_network_getaddresses: getaddrinfo for laravel-protector-mysql_testing-1 failed: Temporary failure in name resolution

and sometimes

SQLSTATE[HY000] [2002] Connection refused

Reply:
As we have already established, the network connection is not properly set up within the package, so it is probably a configuration error. You probably have to work that out with Jörn if you cannot solve it yourself.

Reply:
it simply works when i comment your line of code

Reply:
That's not really a solution, right?

Reply:
didnt mean to imply that

Reply:
We now make sure we are using the correct database by specifying $this->connection when we check that the connected database is MariaDB.